HW_2.py

```diff
@@ -0,0 +1,164 @@
+from pydantic import BaseModel
+from contextlib import contextmanager
+from abc import ABC, abstractmethod
+
+@contextmanager
+def file_opener(filename: str, mode: str):
+    file = open(filename, mode)
+    yield file
+    file.close()
+
+class BookModel(BaseModel):
+    book_name: str
+    author: str
+    year: int
+
+
+class AbstractPublication(ABC):
+    @abstractmethod
+    def get_info(self):
+        raise NotImplementedError
+
+class Book(AbstractPublication):
+    def __init__(self, model: BookModel):
+        self._model = model
+        self.book_name = model.book_name
+        self.author = model.author
+        self.year = model.year
+
+    def __str__(self) -> str:
+        return f"'{self._model.book_name}' {self._model.year} {self._model.author}"
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    def get_info(self) -> str:
+        return f'Book title: {self._model.book_name}\nAuthor: {self._model.author}\nYear:{self._model.year}'
+
+
+class Magazine(Book):
+    def __init__(self, model: BookModel):
+        super().__init__(model)
+
+    def get_info(self) -> str:
+        return f'Magazine title: {self._model.book_name}\nAuthor: {self._model.author}\nYear:{self._model.year}'
```

Comment:
Good that you shown different implementation for Book and Magazine class. But all string except first word is the same.
So you can create additional attribute like _info_prefix, which will equal 'Book' in case of class Book, and 'Magazine' in case of Magazine. It can be added to abstract class with value None too.
And in get_info you'll return smth like f'{self._info_prefix} title: .....'